content/en/docs/reference/setup-tools/kubeadm/kubeadm-upgrade.md

```diff
@@ -28,6 +28,13 @@ automatically renew the kubeadm managed certificates on this node, including tho
 To opt-out, it is possible to pass the flag `--certificate-renewal=false`. For more details about certificate
 renewal see the [certificate management documentation](docs/tasks/administer-cluster/kubeadm/kubeadm-certs).
 
+{{< note >}}
+The commands `kubeadm upgrade apply` and `kubeadm upgrade plan` have the legacy `--config`
+flag which makes it possible to reconfigure the cluster, while performing planning or upgrade of that particular
+control-plane node. Usage of the `--config` flag is not recommended and can lead to unexpected results.
```

Comment:
I think we should document when this workaround works and how to use it, because ATM this is the best/easiest alternative in the user's hands (e.g updating contol-plane static pods; the command should be run on all the controlplane node)
Usage of the `--config` flag should not be recommended for the other use cases.
+1 for hinting about replacement with a better solution in future releases

Reply:
nit: “Usage of the `--config` flag is not recommended and can lead to unexpected results.” is using the passive voice / doesn't [address the reader as “you”](https://kubernetes.io/docs/contribute/style/style-guide/#address-the-reader-as-you).

Reply:
@fabriziopandini i don't even know when `--config` works and when it's dangerous. we don't have tests for it. unless we enumerate the list of viable usages, i suggest that we keep this note as is.

@sftim suggestions for rewording?

Reply:
As far as I know, it works fine for changes to extra-args for control plane components, while I'm not sure about everything else.

But, If overall we feel we don't know when the flag it is dangerous then might be we should give a more generic warning "Please be aware that the upgrade workflow was not designed for this scenario and there were reports of unexpected results"

Reply:
> As far as I know, it works fine for changes to extra-args for control plane components, while I'm not sure about everything else.

yes, but for single CP scenarios only.
for multi-CP changes do not automatically propagate.

> "Please be aware that the upgrade workflow was not designed for this scenario and there were reports of unexpected results"

i can update the comment.